datafusion/expr/src/udf.rs

```diff
@@ -89,6 +91,34 @@ impl ScalarUDF {
             signature: signature.clone(),
             return_type: return_type.clone(),
             fun: fun.clone(),
+            aliases: vec![],
+        }
+    }
+
+    /// Create a new ScalarUDF with aliases
+    pub fn new_with_aliases(
+        name: &str,
+        signature: &Signature,
+        return_type: &ReturnTypeFunction,
+        fun: &ScalarFunctionImplementation,
+        aliases: Vec<String>,
+    ) -> Self {
+        Self {
+            name: name.to_owned(),
+            signature: signature.clone(),
+            return_type: return_type.clone(),
+            fun: fun.clone(),
+            aliases,
+        }
+    }
+
+    pub fn with_aliases(self, aliases: Vec<&str>) -> Self {
+        Self {
+            name: self.name,
+            signature: self.signature,
+            return_type: self.return_type,
+            fun: self.fun,
+            aliases: aliases.iter().map(|s| s.to_string()).collect(),
         }
```

Comment:
I think you can write this more concisely like this. It would also be good to add comments:

```suggestion
    /// Adds additional names that can be used to invoke this function, in addition to `name`
    pub fn with_aliases(mut self, aliases: impl IntoIterator<Item = &str>) -> Self {
      self.aliases.extend(aliases.into_iter().map(|s| s.to_string()));
      self
    }
```